src/aead/aead.rs

```diff
@@ -76,37 +76,24 @@ impl OpeningKey {
     pub fn algorithm(&self) -> &'static Algorithm { self.key.algorithm() }
 }
 
-/// Authenticates and decrypts (&ldquo;opens&rdquo;) data in place.
-///
-/// The input is `in_out[in_prefix_len..]`; i.e. the input is the part of
-/// `in_out` after the prefix. When `open_in_place` returns `Ok(out_len)`, the
-/// decrypted output is `in_out[..out_len]`; i.e. the output has been written
-/// over the top of the prefix and the input. To put it a different way, the
-/// output overwrites the input, shifted by `in_prefix_len` bytes. To have the
-/// output overwrite the input without shifting, pass 0 as `in_prefix_len`.
-/// (The input/output buffer is expressed this way because Rust's type system
-/// does not allow us to have two slices, one mutable and one immutable, that
-/// reference overlapping memory at the same time.)
+/// Authenticates and decrypts (&ldquo;opens&rdquo;) data in place. When
+/// `open_in_place` returns `Ok(plaintext)`, the decrypted output is
+/// `plaintext`, which is a slice of `in_out`.
 ///
 /// C analog: `EVP_AEAD_CTX_open`
 ///
 /// Go analog: [`AEAD.Open`](https://golang.org/pkg/crypto/cipher/#AEAD)
-pub fn open_in_place(key: &OpeningKey, nonce: &[u8], in_prefix_len: usize,
-                     in_out: &mut [u8], ad: &[u8])
-                     -> Result<usize, error::Unspecified> {
+pub fn open_in_place<'a>(key: &OpeningKey, nonce: &[u8], ad: &[u8],
```

Comment:
> because the caller would need to split `ciphertext + tag`

Why split? When they received it from the `seal_in_place()` function, it was already separate, wasn't it?

I think I can follow your argument, on the assumption that the caller still stores the `ciphertext + tag` in a single place, even though the API has now changed. All (4?) current users of ring's AEAD API would probably do that.
For new users, some might still do that. After all, ciphertext and tag somehow belong together.

But other users could find simpler ways to keep ciphertext and tag separate. Like a `struct Message { sender, time, ..., ciphertext: Vec<u8>, tag: Vec<u8>, ... }`, which they can even serialize to JSON or XML and read with their AEAD implementation in Javascript or elsewhere. For these users, it would be weird if they had to concatenate the two.
Then again, these users prefer the out-of-place API, because they wouldn't want their `message.ciphertext` field to be overwritten. Maybe my example isn't the best, but I hope you see my point.